contracts/exchange/api/BaseContract.sol

```diff
@@ -10,10 +10,13 @@ import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.
 contract BaseContract is ReentrancyGuardUpgradeable {
   State internal state;
 
+  // This line below is replaced when we run `yarn compile`
+  uint private constant chainID = 271; // REPLACE ME with the correctchainID;
+
   bytes32 private constant EIP712_DOMAIN_TYPEHASH =
     keccak256("EIP712Domain(string name,string version,uint256 chainId)");
   bytes32 private constant DOMAIN_HASH =
-    keccak256(abi.encode(EIP712_DOMAIN_TYPEHASH, keccak256(bytes("GRVT Exchange")), keccak256(bytes("0")), 1));
+    keccak256(abi.encode(EIP712_DOMAIN_TYPEHASH, keccak256(bytes("GRVT Exchange")), keccak256(bytes("0")), chainID));
```

Comment:
For smart contract it might be cleaner to use `block.chainid`(https://docs.soliditylang.org/en/latest/units-and-global-variables.html#block-and-transaction-properties).

Reply:
Agreed. I made the change